include/seastar/core/sstring.hh

```diff
@@ -65,6 +65,7 @@ class basic_sstring {
         struct external_type {
             char_type* str;
             Size size;
+            bool is_ref;//all non-managed non-copy string references should set this flag to true
```

Comment:
Regardless of whether this patch is needed at all (see my other comment) please note you shouldn't just add something to "external type" - it will make it longer, and longer than "internal_type", and waste space. What you can do is to replace the old "pad" with a single-byte flag.